rust/src/treefile.rs

```diff
@@ -2755,7 +2755,11 @@ impl TreeComposeConfig {
     fn hasher_update(&self, hasher: &mut glib::Checksum) -> Result<()> {
         // don't use pretty mode to increase the chances of a stable serialization
         // https://github.com/projectatomic/rpm-ostree/pull/1865
-        hasher.update(serde_json::to_vec(self)?.as_slice());
+        let j_str = serde_json::to_string(self)?;
+        let mut j_obj: TreeComposeConfig = serde_json::from_str(&j_str)?;
+        j_obj.base.ostree_layers = None;
```

Comment:
Let's add a comment above this for why we're doing this.  How about e.g.:
`// We don't want to hash in the names of the layers, because only their content is relevant` or so?